src/FacebookMessage.php

```diff
@@ -58,13 +90,43 @@ public function to($recipient)
      * Notification text.
      *
      * @param $text
+     * @throws CouldNotCreateMessage
      *
      * @return $this
      */
     public function text($text)
     {
-        $this->text = $text;
+        if (!mb_strlen($text) > 320)
+            $this->text = $text;
+        else
+            throw CouldNotCreateMessage::textTooLong();
+        $this->has_text = true;
+        return $this;
+    }
 
+    /**
+     * Add Attachment
+     *
+     * @param $attachment_type
+     * @param $url
+     * @throws CouldNotCreateMessage
+     *
+     * @return $this
+     */
+    public function attach($attachment_type, $url)
+    {
+        if (in_array($attachment_type, [AttachmentType::FILE, AttachmentType::IMAGE,
+            AttachmentType::VIDEO, AttachmentType::AUDIO]))
```

Comment:
Can we move that array above the method like so? :

```php
$attachment_types = [AttachmentType::FILE, AttachmentType::IMAGE, AttachmentType::VIDEO, AttachmentType::AUDIO];

if (in_array($attachment_type, $attachment_types))
```

It's for readability purposes.